src/cpp/rtps/RTPSDomain.cpp

```diff
@@ -434,12 +434,13 @@ RTPSParticipantImpl* RTPSDomainImpl::find_local_participant(
 RTPSReader* RTPSDomainImpl::find_local_reader(
         const GUID_t& reader_guid)
 {
-    std::lock_guard<std::mutex> guard(RTPSDomain::m_mutex);
+    std::unique_lock<std::mutex> lock(RTPSDomain::m_mutex);
     for (const RTPSDomain::t_p_RTPSParticipant& participant : RTPSDomain::m_RTPSParticipants)
     {
         if (participant.second->getGuid().guidPrefix == reader_guid.guidPrefix)
         {
             // Participant found, forward the query
+            lock.unlock();
```

Comment:
This lock cannot be released here, as it is protecting m_RTPSParticipants and the participant reference.